modules/expo-bluesky-video-player/ios/PlayerItemManager.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import AVKit
+import CryptoKit
+
+class PlayerItemManager {
+  static let shared = PlayerItemManager()
+  
+  private var prefetchQueue: [String] = []
+  private var assets: [String:AVURLAsset] = [:]
+  
+  func getAsset(source: String) -> AVURLAsset? {
+    let path = createPath(source)
+    
+    if let asset = assets[source] {
+      return asset
+    }
+
+    if FileManager.default.fileExists(atPath: path.path) {
+      guard let asset = AVAsset(url: path) as? AVURLAsset else {
+        return nil
+      }
+      self.assets[source] = asset
+      return asset
+    } else if let url = URL(string: source) {
+      guard let asset = AVAsset(url: url) as? AVURLAsset else {
+        return nil
+      }
+      self.assets[source] = asset
+      return asset
```

Comment:
These will always be of `AVURLAsset` but I really don't like doing `as!`, just in case.